server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java

```diff
@@ -117,31 +125,26 @@ public InputStreamResource downloadLog(Long scheduleTaskId, OdcTaskLogLevel leve
 
     @SneakyThrows
     public String getFullLogDownloadUrl(Long scheduleId, Long scheduleTaskId, OdcTaskLogLevel level) {
-        if (ObjectUtil.isNull(cloudObjectStorageService)) {
-            log.warn("cloud object storage service is null.");
+        ScheduleTaskEntity taskEntity = scheduleTaskService.nullSafeGetById(scheduleTaskId);
+        Long jobId = taskEntity.getJobId();
+        if (!taskFrameworkEnabledProperties.isEnabled() || jobId == null) {
             return String.format(DOWNLOAD_LOG_URL_PATTERN, scheduleId, scheduleTaskId);
         }
-        if (cloudObjectStorageService.supported()) {
-            ScheduleTaskEntity taskEntity = scheduleTaskService.nullSafeGetById(scheduleTaskId);
-            Long jobId = taskEntity.getJobId();
-            if (jobId == null) {
-                log.warn("job is not exist, may a historical schedule task, scheduleTaskId:{}", scheduleTaskId);
+        JobEntity jobEntity = taskFrameworkService.find(jobId);
+        PreConditions.notNull(jobEntity, "job not found by id " + jobId);
+        if (JobUtils.isK8sRunMode(jobEntity.getRunMode())) {
+            CloudObjectStorageService cloudObjectStorageService = getCloudObjectStorageService(jobEntity);
+            if (!cloudObjectStorageService.supported()) {
                 return StrUtil.EMPTY;
             }
-            JobEntity jobEntity = taskFrameworkService.find(jobId);
-            PreConditions.notNull(jobEntity, "job not found by id " + jobId);
-            if (JobUtils.isK8sRunMode(jobEntity.getRunMode())) {
-                String attributeKey =
-                        OdcTaskLogLevel.ALL.equals(level) ? JobAttributeKeyConstants.LOG_STORAGE_ALL_OBJECT_ID
-                                : JobAttributeKeyConstants.LOG_STORAGE_WARN_OBJECT_ID;
-                Optional<String> objId = taskFrameworkService.findByJobIdAndAttributeKey(jobId, attributeKey);
-                Optional<String> bucketName = taskFrameworkService.findByJobIdAndAttributeKey(jobId,
-                        JobAttributeKeyConstants.LOG_STORAGE_BUCKET_NAME);
-                if (objId.isPresent() && bucketName.isPresent()) {
-                    return cloudObjectStorageService.generateDownloadUrl(objId.get()).toString();
-                }
-            } else {
-                log.warn("cloud object storage service is not support.");
+            String attributeKey =
+                    OdcTaskLogLevel.ALL.equals(level) ? JobAttributeKeyConstants.LOG_STORAGE_ALL_OBJECT_ID
+                            : JobAttributeKeyConstants.LOG_STORAGE_WARN_OBJECT_ID;
+            Optional<String> objId = taskFrameworkService.findByJobIdAndAttributeKey(jobId, attributeKey);
```

Comment:
unessary multiple query for get attributes, may refer JobAttributeRepository.findByJobId().

we may add a `Map<String,String> getJobAttributes(Long jobId)` method in TaskFrameworkService